File: 1/2021.03.24/455.Assign Cookies/java/solution.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int i = g.length - 1, j = s.length - 1, res = 0;
        while (i >= 0 && j >= 0) {
            if (s[j] >= g[i]) {
                res++;
                i--;
                j--;
            } else {
                i--;
            } 
        }
        return res;
    }
}

File: 1/2021.03.24/455.Assign Cookies/python/solution.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort(reverse=True)
        s.sort(reverse=True)
        i = j = res = 0
        while i < len(g) and j < len(s):
            if s[j] >= g[i]:
                res += 1
                i += 1
                j += 1
            else:
                i += 1
        return res
